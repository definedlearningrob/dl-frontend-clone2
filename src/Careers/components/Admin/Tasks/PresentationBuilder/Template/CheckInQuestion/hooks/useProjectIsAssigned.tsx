import { useMemo } from 'react';
import { useParams } from 'react-router-dom';
import { useQuery } from '@apollo/client';
import { isEmpty } from 'lodash-es';

import teamProjectQuery from '@pbl/graphql/student/queries/teamProject';
import projectQuery from '@pbl/graphql/student/queries/project';

import { useRole } from '@shared/hooks/useRole';

export const useProjectIsAssigned = () => {
  const { isStudent } = useRole();
  const { projectId, teamId } = useParams<{ projectId: string; teamId: string }>();

  const properQuery = useMemo(() => (teamId ? teamProjectQuery : projectQuery), [teamId]);

  const queryVariables = useMemo(
    () =>
      teamId
        ? { id: projectId, teamId, track: true, trackPresentation: true }
        : { id: projectId, track: true, trackPresentation: true },
    [teamId]
  );

  const { data, error, loading } = useQuery(properQuery, {
    variables: queryVariables,
    skip: !isStudent,
  });

  if (!isStudent || loading || error) {
    return false;
  }

  return !isEmpty(data?.project?.assignedAt);
};
