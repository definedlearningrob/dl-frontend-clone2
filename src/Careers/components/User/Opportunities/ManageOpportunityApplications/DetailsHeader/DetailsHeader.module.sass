@import '@shared/stylesheets/config'

.link
  +ellipsis
  font-size: $font-size-xs
  color: $color-primary-500
  text-decoration: none
  position: relative

  &:after
    +pseudoelement
    background: $color-neutral-300
    border-radius: 100%
    width: 4px
    height: 4px
    right: -10px
    left: inherit
    top: 7px

  &:hover
    color: $color-primary-600
