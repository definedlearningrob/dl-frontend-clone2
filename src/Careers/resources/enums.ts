export enum CourseType {
  HIGH_SCHOOL = 'HIGH_SCHOOL',
  MIDDLE_SCHOOL = 'MIDDLE_SCHOOL',
}
export enum TagTypes {
  SYSTEM = 'SYSTEM',
  ENTITY = 'ENTITY',
}

export enum ResourceClass {
  ASSIGNMENT_SUBMISSION_FILE = 'ASSIGNMENT_SUBMISSION_FILE',
  ATTACHMENT = 'ATTACHMENT',
  CATALOG = 'CATALOG',
  COURSE = 'COURSE',
  EXTENSION_FIELD = 'EXTENSION_FIELD',
  EXTENSION_FIELD_FILE = 'EXTENSION_FIELD_FILE',
  LESSON = 'LESSON',
  PORTFOLIO_PROJECT = 'PORTFOLIO_PROJECT',
  PORTFOLIO_PROJECT_FILE = 'PORTFOLIO_PROJECT_FILE',
  PUBLIC_RESOURCE = 'PUBLIC_RESOURCE',
  RESUME = 'RESUME',
  SLIDE_BACKGROUND_IMAGE = 'SLIDE_BACKGROUND_IMAGE',
  SLIDE_IMAGE = 'SLIDE_IMAGE',
  SLIDE_VIDEO = 'SLIDE_VIDEO',
  TASK = 'TASK',
  TASK_FILE = 'TASK_FILE',
  TRACK = 'TRACK',
  UNIT = 'UNIT',
  VIDEO = 'VIDEO',
  BADGE = 'BADGE',
  ENTITY = 'ENTITY',
}

export enum AssetType {
  FILE = 'FILE',
  IMAGE = 'IMAGE',
  VIDEO = 'VIDEO',
  DC_LOGO = 'DC_LOGO',
  DC_ICON = 'DC_ICON',
  DL_LOGO = 'DL_LOGO',
  DL_ICON = 'DL_ICON',
}

export enum AssessmentType {
  HIGH_SCHOOL = 'HIGH_SCHOOL',
  MIDDLE_SCHOOL = 'MIDDLE_SCHOOL',
}

export enum AssessmentStatuses {
  COMPLETED = 'COMPLETED',
  FAILED = 'FAILED',
  FINISHED = 'FINISHED',
  IN_PROGRESS = 'IN_PROGRESS',
  NOT_STARTED = 'NOT_STARTED',
  PROCESSING_RESULTS = 'PROCESSING_RESULTS',
}

export enum ConversationContextTypes {
  ASSIGNMENT = 'ASSIGNMENT',
  LESSON = 'LESSON',
  COURSE = 'COURSE',
  CHECK_IN_ANSWER = 'CHECK_IN_ANSWER',
  ANNOUNCEMENT = 'ANNOUNCEMENT',
  OPPORTUNITY = 'OPPORTUNITY',
}

export enum Roles {
  ENTITY_ADMIN = 'ENTITY_ADMIN',
  SALES_ADMIN = 'SALES_ADMIN',
  SYSTEM_ADMIN = 'SYSTEM_ADMIN',
  TEACHER = 'TEACHER',
}

export enum EducationalSettingTypes {
  HIGH_SCHOOL = 'HIGH_SCHOOL',
  MIDDLE_SCHOOL = 'MIDDLE_SCHOOL',
}

export enum ReportLevels {
  USER = 'USER',
  SCHOOL_CLASS = 'SCHOOL_CLASS',
  ENTITY = 'ENTITY',
}

export enum UploadReportStatuses {
  NOT_STARTED = 'NOT_STARTED',
  IN_PROGRESS = 'IN_PROGRESS',
  COMPLETED = 'COMPLETED',
  FAILED = 'FAILED',
}

export enum ArchivableStatusTypes {
  ACTIVE = 'ACTIVE',
  ALL = 'ALL',
  ARCHIVED = 'ARCHIVED',
}

export enum ContentStatusesTypes {
  DRAFT = 'DRAFT',
  PUBLISHED = 'PUBLISHED',
}

export enum ServiceTypes {
  ALL = 'ALL',
  CAREERS = 'CAREERS',
  LEARNING = 'LEARNING',
}

export enum AssignmentSubmissionStatus {
  DRAFT = 'DRAFT',
  SUBMITTED = 'SUBMITTED',
}

export enum OPPORTUNITY_APPLICATION_STATUS {
  ACCEPTED = 'ACCEPTED',
  FINISHED = 'FINISHED',
  PENDING = 'PENDING',
  REJECTED = 'REJECTED',
  STARTED = 'STARTED',
  EXPIRED = 'EXPIRED',
}

export enum VIRTUAL_INTERNSHIP_STATUS {
  NOT_STARTED = 'NOT_STARTED',
  IN_PROGRESS = 'IN_PROGRESS',
  COMPLETED = 'COMPLETED',
}

export enum OPPORTUNITY_TYPE {
  APPRENTICESHIP = 'APPRENTICESHIP',
  CLINICAL_EXPERIENCE = 'CLINICAL_EXPERIENCE',
  INTERNSHIP = 'INTERNSHIP',
  JOB_SHADOW = 'JOB_SHADOW',
  OTHER = 'OTHER',
  PRACTICUM = 'PRACTICUM',
  PRE_APPRENTICESHIP = 'PRE_APPRENTICESHIP',
  VIRTUAL_INTERNSHIP = 'VIRTUAL_INTERNSHIP',
  WORKPLACE_TOUR = 'WORKPLACE_TOUR',
  JOB = 'JOB',
  COLLEGE_VISIT = 'COLLEGE_VISIT',
  SPEAKER = 'SPEAKER',
}

export enum VISIBILITY_SCOPE {
  ALL = 'ALL',
  ENTITY = 'ENTITY',
}

export enum INSTITUTION_TYPES {
  PRV_NFP_LT2 = 'Private not-for-profit, less-than 2-year',
  PRV_NFP_2 = 'Private not-for-profit, 2-year',
  PRV_NFP_4 = 'Private not-for-profit, 4-year or above',
  PRV_FP_LT2 = 'Private for-profit, less-than 2-year',
  PRV_FP_2 = 'Private for-profit, 2-year',
  PRV_FP_4 = 'Private for-profit, 4-year or above',
  PUB_LT2 = 'Public, less-than 2-year',
  PUB_2 = 'Public, 2-year',
  PUB_4 = 'Public, 4-year or above',
}

export enum US_STATES {
  ALASKA = 'Alaska',
  ALABAMA = 'Alabama',
  ARKANSAS = 'Arkansas',
  AMERICAN_SAMOA = 'American Samoa',
  ARIZONA = 'Arizona',
  CALIFORNIA = 'California',
  COLORADO = 'Colorado',
  CONNECTICUT = 'Connecticut',
  DISTRICT_OF_COLUMBIA = 'District Of Columbia',
  DELAWARE = 'Delaware',
  FLORIDA = 'Florida',
  GEORGIA = 'Georgia',
  GUAM = 'Guam',
  HAWAII = 'Hawaii',
  IOWA = 'Iowa',
  IDAHO = 'Idaho',
  ILLINOIS = 'Illinois',
  INDIANA = 'Indiana',
  KANSAS = 'Kansas',
  KENTUCKY = 'Kentucky',
  LOUISIANA = 'Louisiana',
  MASSACHUSETTS = 'Massachusetts',
  MARYLAND = 'Maryland',
  MAINE = 'Maine',
  MICHIGAN = 'Michigan',
  MINNESOTA = 'Minnesota',
  MISSOURI = 'Missouri',
  MISSISSIPPI = 'Mississippi',
  MONTANA = 'Montana',
  NORTH_CAROLINA = 'North Carolina',
  NORTH_DAKOTA = 'North Dakota',
  NEBRASKA = 'Nebraska',
  NEW_HAMPSHIRE = 'New Hampshire',
  NEW_JERSEY = 'New Jersey',
  NEW_MEXICO = 'New Mexico',
  NEVADA = 'Nevada',
  NEW_YORK = 'New York',
  OHIO = 'Ohio',
  OKLAHOMA = 'Oklahoma',
  OREGON = 'Oregon',
  PENNSYLVANIA = 'Pennsylvania',
  PUERTO_RICO = 'Puerto Rico',
  RHODE_ISLAND = 'Rhode Island',
  SOUTH_CAROLINA = 'South Carolina',
  SOUTH_DAKOTA = 'South Dakota',
  TENNESSEE = 'Tennessee',
  TEXAS = 'Texas',
  UTAH = 'Utah',
  VIRGINIA = 'Virginia',
  VIRGIN_ISLANDS = 'Virgin Islands',
  VERMONT = 'Vermont',
  WASHINGTON = 'Washington',
  WISCONSIN = 'Wisconsin',
  WEST_VIRGINIA = 'West Virginia',
  WYOMING = 'Wyoming',
}

export enum APPLICATIONS_TYPE {
  COMMON_APP = 'COMMON_APP',
  DIRECT = 'DIRECT',
}

export enum RECOMMENDER_TYPE {
  TEACHER = 'TEACHER',
  COUNSELOR = 'COUNSELOR',
}

export enum INSTITUTION_APPLICATION_STATUS {
  NOT_STARTED = 'NOT_STARTED',
  IN_PROGRESS = 'IN_PROGRESS',
  SUBMITTED = 'SUBMITTED',
  COMPLETED = 'COMPLETED',
  DOWNLOADED = 'DOWNLOADED',
}
export enum COMMON_APP_FORM_STATUS {
  NOT_STARTED = 'NOT_STARTED',
  IN_PROGRESS = 'IN_PROGRESS',
  SUBMITTED = 'SUBMITTED',
  COMPLETED = 'COMPLETED',
  DOWNLOADED = 'DOWNLOADED',
}

export enum APPLICATION_FORM_TYPE {
  COUNSELOR_EARLY_DECISION_2 = 'Counselor early decision 2',
  COUNSELOR_EARLY_DECISION = 'Counselor early decision',
  COUNSELOR_FEE_WAIVER = 'Counselor fee waiver',
  COUNSELOR_FINAL_REPORT = 'Counselor final report',
  COUNSELOR_MIDYEAR_REPORT = 'Counselor midyear report',
  COUNSELOR_OPTIONAL_REPORT_2 = 'Counselor optional report 2',
  COUNSELOR_OPTIONAL_REPORT = 'Counselor optional report',
  COUNSELOR_PROFILE = 'Counselor profile',
  COUNSELOR_RECOMMENDATION = 'Counselor recommendation',
  COUNSELOR_SCHOOL_REPORT = 'Counselor school report',
  COUNSELOR_SECONDARY_REPORT = 'Counselor secondary report',
  TEACHER_PROFILE = 'Teacher profile',
  TEACHER_RECOMMENDATION = 'Teacher recommendation',
  TEACHER_EVALUATION = 'Teacher evaluation',
}

export enum SIZE_TYPES {
  VERY_SMALL = 'VERY_SMALL', // Size for very small institutions
  SMALL = 'SMALL', // Size for small institutions
  MEDIUM = 'MEDIUM', // Size for medium institutions
  LARGE = 'LARGE', // Size for large institutions
  VERY_LARGE = 'VERY_LARGE', // Size for very large institutions
}

export enum COST_RANGES {
  LESS_THAN_5000 = 'LESS_THAN_5000', // Institution cost value between 0 and 5000
  FROM_5001_TO_10000 = 'FROM_5001_TO_10000', //Institution cost value between 5001 and 10000
  FROM_10001_TO_15000 = 'FROM_10001_TO_15000', //Institution cost value between 10001 and 15000
  FROM_15001_TO_20000 = 'FROM_15001_TO_20000', //Institution cost value between 15001 and 20000
  MORE_THAN_20000 = 'MORE_THAN_20000', //Institution cost value higher than 20000
}

export enum LOCALE_STORAGE_KEY {
  APPLICATION_REFRESH_TIME = 'APPLICATION_REFRESH_TIME',
}

export enum CAREER_REVIEW_SURVEY_ANSWER_CONTEXT_TYPES {
  COURSE = 'COURSE',
  VIRTUAL_INTERNSHIP = 'VIRTUAL_INTERNSHIP',
  ASSESSMENT = 'ASSESSMENT',
}

export enum SYNC_STATUS {
  COMPLETED = 'COMPLETED',
  IN_PROGRESS = 'IN_PROGRESS',
  FAILED = 'FAILED',
}

export enum ASSIGNMENT_SUBMISSION_TYPES {
  COURSE = 'COURSE',
  VIRTUAL_INTERNSHIP = 'VIRTUAL_INTERNSHIP',
}

export enum ASSESSMENT_TYPES {
  HIGH_SCHOOL = 'HIGH_SCHOOL',
  MIDDLE_SCHOOL = 'MIDDLE_SCHOOL',
}

export enum ASSESSMENT_ATTEMPT_STATUS {
  IN_PROGRESS = 'IN_PROGRESS',
  FINISHED = 'FINISHED',
  REJECTED = 'REJECTED',
  PROCESSING_RESULTS = 'PROCESSING_RESULTS',
  FAILED = 'FAILED',
}

export enum ASSESSMENT_STATUSES {
  COMPLETED = 'COMPLETED',
  NOT_STARTED = 'NOT_STARTED',
  IN_PROGRESS = 'IN_PROGRESS',
}
