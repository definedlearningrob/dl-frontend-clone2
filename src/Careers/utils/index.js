const isObject = (object) => object !== null && typeof object === 'object';

export const isEqual = (object1, object2) => {
  if (object1 === object2) return true;

  if (
    object1 === null ||
    object2 === null ||
    typeof object1 !== 'object' ||
    typeof object2 !== 'object'
  )
    return false;

  const keys1 = Object.keys(object1);
  const keys2 = Object.keys(object2);

  if (keys1.length !== keys2.length) {
    return false;
  }

  for (let key of keys1) {
    const val1 = object1[key];
    const val2 = object2[key];
    const areObjects = isObject(val1) && isObject(val2);
    if (
      // eslint-disable-next-line
      areObjects && !isEqual(val1, val2) ||
      // eslint-disable-next-line
      !areObjects && val1 !== val2
    ) {
      return false;
    }
  }

  return true;
};

export const getByStringKey = (object, key) => {
  const keys = key.split('.');

  return keys.reduce((acc = {}, key) => acc[key], object);
};
