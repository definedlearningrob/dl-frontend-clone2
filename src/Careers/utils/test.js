/* eslint-disable react/prop-types */
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { createMemoryHistory } from 'history';
import { I18nextProvider } from 'react-i18next';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import { Router, Route } from 'react-router-dom';

import i18n from '@dc/i18n';

export function renderWithRouter(
  ui,
  { route = '/', routePath = '/', history = createMemoryHistory({ initialEntries: [route] }) } = {}
) {
  const Wrapper = ({ children }) => (
    <Router history={history}>
      <I18nextProvider i18n={i18n}>
        <Route path={routePath} render={() => children} />
      </I18nextProvider>
    </Router>
  );

  return {
    ...render(ui, { wrapper: Wrapper }),
    history,
  };
}

export function renderWithReduxProvider(
  ui,
  {
    initialState = {
      session: {
        loginError: {},
        user: null,
      },
    },
    store = configureMockStore([thunk])(initialState),
  } = {}
) {
  const Wrapper = ({ children }) => (
    <Provider store={store}>
      <I18nextProvider i18n={i18n}>{children}</I18nextProvider>
    </Provider>
  );

  return {
    ...render(ui, { wrapper: Wrapper }),
  };
}

export function renderWithRouterAndReduxProvider(
  ui,
  {
    route = '/',
    history = createMemoryHistory({ initialEntries: [route] }),
    initialState = {
      session: {
        loginError: {},
        user: {},
      },
    },
    store = configureMockStore([thunk])(initialState),
  } = {},
  ...options
) {
  const Wrapper = ({ children }) => (
    <Provider store={store}>
      <Router history={history}>
        <I18nextProvider i18n={i18n}>{children}</I18nextProvider>
      </Router>
    </Provider>
  );

  return {
    ...render(ui, { wrapper: Wrapper, ...options }),
    history,
  };
}

export function renderWithI18N(ui) {
  const Wrapper = ({ children }) => <I18nextProvider i18n={i18n}>{children}</I18nextProvider>;

  return {
    ...render(ui, { wrapper: Wrapper }),
  };
}
