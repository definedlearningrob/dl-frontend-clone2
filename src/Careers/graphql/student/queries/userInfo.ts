import { gql } from '@apollo/client';

import { EducationalSettingTypes, US_STATES } from '@dc/resources/enums';
import { TSyncStatus } from '@dc/resources/types';

export type TCounselor = {
  email: string;
  firstName: string;
  lastName: string;
  uuid: string;
};

export type TCommonAppData = {
  canSelectCounselor: boolean;
  canChangeCounselor: boolean;
  connectionUrl: string;
  hasAccountConnected: boolean;
  currentCounselor: TCounselor | null;
  hasCounselorInvited: boolean;
  hasFerpaSigned: boolean;
  syncStatus: TSyncStatus | null;
};

export type TStudentInfoSettings = {
  assessmentEnabled: boolean;
  assessmentType: EducationalSettingTypes;
  onboardingEnabled: boolean;
  selfEvaluationEnabled: boolean;
};

export type TStudentInfo = {
  currentSchoolYear: number;
  email: string;
  hasCompletedAssessment: boolean;
  hasCompletedOnboarding: boolean;
  hasUnreadConversation: boolean;
  hasAccessToPbl: boolean;
  firstName: string;
  isImpersonated: boolean;
  hasOpportunitiesEnabled: boolean;
  hasPlans: boolean;
  lastName: string;
  logoUrl: string;
  iconUrl: string;
  settings: TStudentInfoSettings;
  state: keyof typeof US_STATES;
  unreadAnnouncementsCount: number;
  postSecondaryApplicationsEnabled: boolean;
  unreadNotificationsCount: number;
  username: string;
  uuid: string;
  commonAppData: TCommonAppData;
  welcomeMessage: string;
};

export type TStudentInfoData = {
  userInfo: TStudentInfo;
};

export default gql`
  query DcStudentInfo {
    userInfo {
      settings {
        assessmentEnabled
        assessmentType
        onboardingEnabled
        selfEvaluationEnabled
      }
      commonAppData {
        canSelectCounselor
        hasCounselorInvited
        canChangeCounselor
        connectionUrl
        hasAccountConnected
        syncStatus {
          lastSyncedAt
          status
        }
        currentCounselor {
          email
          firstName
          lastName
          uuid
        }
        hasFerpaSigned
      }
      currentSchoolYear
      email
      hasCompletedAssessment
      hasCompletedOnboarding
      hasUnreadConversation
      hasOpportunitiesEnabled
      hasPlans
      postSecondaryApplicationsEnabled
      hasAccessToPbl
      firstName
      isImpersonated
      lastName
      logoUrl
      iconUrl
      state
      unreadAnnouncementsCount
      unreadNotificationsCount
      username
      uuid
      welcomeMessage
    }
  }
`;
