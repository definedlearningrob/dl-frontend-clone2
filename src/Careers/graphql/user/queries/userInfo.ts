import { gql } from '@apollo/client';

import { Roles } from '@dc/resources/enums';
import { TSyncStatus } from '@dc/resources/types';

import { ReportType } from '@shared/resources/enums';

import { TUserPermissions } from './user';

type TUserInfoEntityNode = {
  uuid: string;
  settings: {
    postSecondaryApplicationsEnabled: boolean;
    classManagementEnabled: boolean;
    schoolYearStartDate: {
      day: number;
      month: number;
    };
  };
  reportTypes: ReportType[];
};

export type TCommonAppData = {
  hasRecommenderInvitation: boolean;
  hasTeacherInvitation: boolean;
  hasCounselorInvitation: boolean;
  hasCounselorProfileFormCompleted: boolean;
  hasTeacherProfileFormCompleted: boolean;
  syncStatus: TSyncStatus | null;
};

export type TUserInfo = {
  availableReportTypes: ReportType[];
  commonAppData: TCommonAppData;
  currentSchoolYear: number;
  email: string;
  logoUrl?: string;
  iconUrl?: string;
  entities: {
    nodes: TUserInfoEntityNode[];
  };
  firstName: string;
  hasUnreadConversation: boolean;
  hasOpportunitiesEnabled: boolean;
  hasAccessToPbl: boolean;
  lastName: string;
  isImpersonated: boolean;
  permissions: TUserPermissions;
  role: Roles;
  username: string;
  uuid: string;
  welcomeMessage: string;
};

export type TUserInfoData = {
  userInfo: TUserInfo;
};

export default gql`
  query DcUserInfo {
    userInfo {
      availableReportTypes
      email
      logoUrl
      iconUrl
      commonAppData {
        hasRecommenderInvitation
        hasTeacherInvitation
        hasCounselorInvitation
        hasCounselorProfileFormCompleted
        hasTeacherProfileFormCompleted
        syncStatus {
          lastSyncedAt
          status
        }
      }
      entities(page: 1, perPage: 1) {
        nodes {
          uuid
          settings {
            postSecondaryApplicationsEnabled
            classManagementEnabled
            schoolYearStartDate {
              day
              month
            }
          }
          reportTypes
        }
      }
      currentSchoolYear
      firstName
      hasUnreadConversation
      hasOpportunitiesEnabled
      hasAccessToPbl
      lastName
      isImpersonated
      permissions {
        wblAdmin
        counselor
        canImpersonate
        canBrowseReports
      }
      role
      username
      uuid
      welcomeMessage
    }
  }
`;
