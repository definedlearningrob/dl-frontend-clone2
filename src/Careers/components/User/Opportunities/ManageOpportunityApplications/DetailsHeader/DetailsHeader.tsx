import { useTranslation } from 'react-i18next';
import { useParams } from 'react-router-dom';

import { TOpportunityApplication } from '@dc/resources/types';
import { ConversationContextTypes } from '@dc/resources/enums';

import { ReactComponent as SendOutlinedIcon } from '@shared/svg/send_outlined.svg';
import { Tooltip } from '@shared/components/Tooltip';
import { formatDateTime, parseDate } from '@shared/utils/date';
import DeprecatedIconButton from '@shared/components/DeprecatedIconButton/DeprecatedIconButton';
import Link from '@shared/components/Link';
import { useMessaging } from '@shared/hooks/useMessaging';

import styles from './DetailsHeader.module.sass';

type Props = {
  selectedApplication: TOpportunityApplication;
  opportunityName: string;
};

export const DetailsHeader = ({ selectedApplication, opportunityName }: Props) => {
  const { t } = useTranslation();
  const { id } = useParams<{ id: string }>();
  const { messagingState, setMessagingState } = useMessaging();

  const applicationHeaderClassName =
    'flex justify-between items-center col-gap-md py-sm px-base border-b border-b-neutral-300 xxxl:py-base xxxl:px-md';

  const toggleMessagingModal = () => {
    setMessagingState({
      ...messagingState,
      show: !messagingState.show,
      receiver: {
        ...selectedApplication.student,
        name: selectedApplication.student.fullName,
      },
      actionContext: {
        id,
        type: ConversationContextTypes.OPPORTUNITY,
        title: opportunityName,
      },
    });
  };

  return (
    <div className={applicationHeaderClassName}>
      <div>
        <h4 className='text-sm mb-xxs xxxl:text-base'>
          {t('opportunityManageApplications.detailsHeader.heading')}
        </h4>
        <div className='flex items-center gap-xs'>
          <Link
            className={styles.link}
            linkClassName='inline-block'
            to={`/students/${selectedApplication.student.uuid}/portfolio`}
            variant='link'>
            {selectedApplication.student.fullName}
          </Link>
          <div className='h-xxs w-xxs bg-neutral-300 rounded-lg' />
          <span className='whitespace-nowrap text-xs'>
            <Tooltip
              className='inline'
              message={formatDateTime(selectedApplication.appliedAt, {
                withTime: true,
              })}>
              {parseDate(selectedApplication.appliedAt)}
            </Tooltip>
          </span>
        </div>
      </div>
      <DeprecatedIconButton
        aria-label={t('messaging.new.send')}
        className='p-xs'
        icon={<SendOutlinedIcon />}
        iconSize='sm'
        size='md'
        square={true}
        variant='primary'
        onClick={toggleMessagingModal}
      />
    </div>
  );
};
