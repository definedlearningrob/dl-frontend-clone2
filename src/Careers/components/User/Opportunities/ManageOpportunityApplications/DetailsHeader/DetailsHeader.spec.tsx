import userEvent from '@testing-library/user-event';
import { screen } from '@testing-library/dom';
import { MockedProvider } from '@apollo/client/testing';

import { renderWithRouterAndReduxProvider } from '@dc/utils/test';
import { DetailsHeader } from '@dc/components/User/Opportunities/ManageOpportunityApplications/DetailsHeader/DetailsHeader';
import { OPPORTUNITY_APPLICATION_STATUS } from '@dc/resources/enums';
import { TUserInfo } from '@dc/graphql/user/queries/userInfo';

import { MessagingProvider } from '@shared/hooks/useMessaging';
import MessagingModal from '@shared/components/Messaging/Modal/Modal';

const application = {
  id: '1',
  appliedAt: '2023-05-01',
  updatedAt: '2023-05-02',
  status: OPPORTUNITY_APPLICATION_STATUS.REJECTED,
  answers: [{ id: '1', opportunityQuestionId: '1', answer: 'To sleep better!', step: 1 }],
  lastChangedBy: {
    uuid: '123',
    name: 'Teacher Name',
  },
  student: {
    uuid: '123',
    fullName: 'Applicant Student 1',
    schoolClasses: [{ uuid: 'class-uuid', name: 'Class name' }],
  },
  __typename: 'OpportunityApplication',
};

describe('DetailsHeader', () => {
  it('should render correctly', () => {
    const { container } = renderWithRouterAndReduxProvider(
      <DetailsHeader opportunityName='Opportunity Name' selectedApplication={application} />
    );

    expect(container).toMatchSnapshot();
  });

  it('should allow sending message to given student', async () => {
    renderWithRouterAndReduxProvider(
      <MockedProvider>
        <MessagingProvider refreshUser={jest.fn()} userInfo={{} as TUserInfo}>
          <DetailsHeader opportunityName='Opportunity Name' selectedApplication={application} />
          <MessagingModal />
        </MessagingProvider>
      </MockedProvider>
    );

    userEvent.click(screen.getByLabelText('Send Message'));

    expect(screen.getByRole('dialog')).toBeVisible();
  });
});
