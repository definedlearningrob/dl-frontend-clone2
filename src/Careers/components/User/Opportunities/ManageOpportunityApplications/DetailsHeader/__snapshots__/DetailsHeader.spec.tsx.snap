// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailsHeader should render correctly 1`] = `
<div>
  <div
    class="flex justify-between items-center col-gap-md py-sm px-base border-b border-b-neutral-300 xxxl:py-base xxxl:px-md"
  >
    <div>
      <h4
        class="text-sm mb-xxs xxxl:text-base"
      >
        Application Details
      </h4>
      <div
        class="flex items-center gap-xs"
      >
        <a
          class="inline-block"
          href="/students/123/portfolio"
        >
          <button
            class="button flex items-center justify-center gap-xs rounded-sm cursor-pointer text-sm font-medium no-underline focus-visible:outline focus-visible:outline-1 focus-visible:outline-offset-[3px] link -link py-xs px-base leading-[22px] text-white"
            data-testid="button"
            type="button"
          >
            <span
              class="items-center flex justify-center gap-xs"
            >
              Applicant Student 1
            </span>
          </button>
        </a>
        <div
          class="h-xxs w-xxs bg-neutral-300 rounded-lg"
        />
        <span
          class="whitespace-nowrap text-xs"
        >
          <span
            class="inline"
            data-state="closed"
          >
            May 1, 2023
          </span>
        </span>
      </div>
    </div>
    <button
      aria-label="Send Message"
      class="icon-button -primary -size-md -square p-xs"
      data-testid="icon-button"
      type="button"
    >
      <div
        class="icon icon-button__icon -sm-size"
        data-testid="icon"
      >
        <div />
      </div>
    </button>
  </div>
</div>
`;
